packages/eth/src/websocket.rs

```diff
@@ -24,10 +30,43 @@ pub struct WebsocketClient {
     contract_caller_address: Address,
 }
 
-#[derive(Debug, Clone)]
-pub struct KmsKeys {
-    pub main_key_arn: String,
-    pub blob_pool_key_arn: Option<String>,
+#[derive(Debug, Clone, PartialEq)]
+pub enum L1Key {
+    Kms(String),
+    Private(String),
+}
+
+impl<'a> serde::Deserialize<'a> for L1Key {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'a>,
+    {
+        let value = String::deserialize(deserializer)?;
+        if let Some(k) = value.strip_prefix("Kms(").and_then(|s| s.strip_suffix(')')) {
```

Comment:
A brief comment on how to setup the key env somewhere would be nice.
My preference would've been to keep separate env vars so their way of use is more obvious, but not worth it at this point